Doc/library/exceptions.rst

```diff
@@ -212,6 +212,12 @@ The following exceptions are the exceptions that are usually raised.
    accidentally caught by code that catches :exc:`Exception` and thus prevent
    the interpreter from exiting.
 
+   .. note::
+
+      Because it can be raised at unpredictable points, a KeyboardInterrupt
+      may, in some circumstances, leave the running program in an inconsistent
+      state. See :ref:`handlers-and-exceptions`.
```

Comment:
This needs clarification that it only applies in cases where `KeyboardInterrupt` is caught.  Otherwise people will become afraid to press Ctrl-C...

I'd also directly mention what to do: "... you should not try to catch KeyboardInterrupt and continue, but let it end the program as quickly as possible, or avoid raising it completely (see ...)"

Reply:
I have updated the pull request.